src/index/base.cpp

```diff
@@ -134,6 +148,25 @@ void BaseIndexNotifications::blockConnected(const interfaces::BlockInfo& block)
 
 void BaseIndexNotifications::blockDisconnected(const interfaces::BlockInfo& block)
 {
+    // During initial sync, ignore validation interface notifications, only
+    // process notifications from sync thread.
+    if (!m_index.m_synced && block.chain_tip) return;
```

Comment:
https://github.com/bitcoin/bitcoin/pull/24230/commits/ac1369650de26f325076340017745c67bdb34900

Is this new behavior or inherited?